### PR TITLE
Chile large update

### PR DESCRIPTION
**Pull request by KJhellico**
- holidays relevance period fixes
- historical holidays added
- special holidays added
- tests for all holidays

### CODE REVIEW

File: holidays/countries/chile.py [hunk 1]
```diff
@@ -64,127 +68,174 @@ class Chile(HolidayBase):
...
-            elif date(year, JUN, 29).weekday() == FRI:
-                self[date(year, JUN, 29) + rd(weekday=MO)] = name
-            else:
-                self[date(year, JUN, 29)] = name
+        self[date(year, MAY, 21)] = "Día de las Glorias Navales [Navy Day]"
+
+        # National Day of Indigenous Peoples
+        if year >= 2021:
+            dt = date(year, JUN, 21)
+            if year >= 2022 and (year - 2022) % 4 > 1:
```
Comment (arkid15r) [inline holidays/countries/chile.py:118]: Could you add a comment explaining what's going on here?

File: test/countries/test_chile.py [hunk 2]
```diff
@@ -9,100 +9,365 @@
...
+            "2020-08-15",
+            "2020-09-18",
+            "2020-09-19",
+            "2020-10-12",
+            "2020-10-31",
+            "2020-11-01",
+            "2020-12-08",
+            "2020-12-25",
+        )
+        self.assertIn("2020-06-07", self.holidays_AP)
```
Comment (arkid15r) [inline test/countries/test_chile.py:348]:
```suggestion
        self.assertHoliday(self.holidays_AP, "2020-06-07")
        self.assertHoliday(self.holidays_NB, "2020-08-20")
```

Comment (KJhellico) [reply test/countries/test_chile.py:348]: Done. Thank you!

File: holidays/countries/chile.py (hunk 1 above)
Comment (KJhellico) [reply holidays/countries/chile.py:118]: It's summer solstice :) For this holiday may be better use code from #828

Comment (KJhellico) [reply holidays/countries/chile.py]: Just updated my code to be fully compliant for whole 21st century.

File: test/countries/test_chile.py [hunk 3]
```diff
@@ -9,100 +9,371 @@
...
+        for year in range(2000, 2020):
+            self.assertIn(
+                "Día del Respeto a la Diversidad",
+                ";".join(Chile(years=year).values()),
+            )
+        for year in range(2020, 2050):
+            self.assertIn(
+                "Día del Descubrimiento de dos Mundos",
+                ";".join(Chile(years=year).values()),
+            )
```
Comment (KJhellico) [inline test/countries/test_chile.py:277]: @arkid15r, maybe you can suggest a better code to check specific holiday presence/absence in year?

Comment (KJhellico) [reply test/countries/test_chile.py:277]: With `get_named `, test for **presence** "Ascensión del Señor" will look like
```
self.assertNotEqual(Chile(years=year).get_named("Ascensión del Señor"), [])
```
?

Comment (arkid15r) [reply test/countries/test_chile.py:277]: `self.assertTrue(Chile(years=year).get_named("Ascensión del Señor"))` would also work.
It's up to you, these tests are going to be refactored soon anyway.

Comment (KJhellico) [reply test/countries/test_chile.py:277]: Ok, thank you. So for now I'll leave it as is.

**Comment by dr-prodigy** [comment]
Thank you @KJhellico for the work (and for fixing conflicts too).. merged! :+1:

**Pull request merged**